Modules/_sqlite/connection.c

```diff
@@ -897,21 +897,121 @@ print_or_clear_traceback(callback_context *ctx)
     }
 }
 
+// Return a borrowed reference to built-in str().
+static PyObject *
+fetch_str_builtin(callback_context *ctx)
+{
+    pysqlite_state *state = ctx->state;
+    if (state->bltin_str != NULL) {
+        return state->bltin_str;
+    }
+
+    /* Both PyEval_GetBuiltins and PyDict_GetItemString return
+     * borrowed references. */
+    PyObject *bltins = PyEval_GetBuiltins();
+    if (bltins == NULL) {
+        return NULL;
+    }
+    PyObject *str = PyDict_GetItemString(bltins , "str");
+    if (str == NULL) {
+        return NULL;
+    }
+
+    /* Store a strong reference in the module state;
+     * return a borrowed reference to the caller. */
+    state->bltin_str = Py_NewRef(str);
+    return str;
+}
+
+static PyObject *
+get_exception_message(callback_context *ctx, PyObject *exc)
```

Comment:
Why not simply use `PyObject_Str()`?

Reply:
Oh, TIL! Thanks, that is a convenient C API.